src/plugins/vis_type_pie/public/pie_component.tsx

```diff
@@ -0,0 +1,342 @@
+  const metricFieldFormatter = services.fieldFormats.deserialize(
+    visParams.dimensions.metric.format
+  );
+  const splitChartFormatter = visParams.dimensions.splitColumn
+    ? services.fieldFormats.deserialize(visParams.dimensions.splitColumn[0].format)
+    : visParams.dimensions.splitRow
+    ? services.fieldFormats.deserialize(visParams.dimensions.splitRow[0].format)
+    : undefined;
+  const percentFormatter = services.fieldFormats.deserialize({
+    id: 'percent',
+    params: {
+      pattern: `0,0.[${'0'.repeat(visParams.labels.percentDecimals ?? DEFAULT_PERCENT_DECIMALS)}]%`,
+    },
+  });
+
+  const { bucketColumns, metricColumn } = useMemo(() => getColumns(visParams, visData), [
+    visData,
+    visParams,
+  ]);
+
+  const layers = useMemo(
+    () =>
+      getLayers(
+        bucketColumns,
+        visParams,
+        props.uiState?.get('vis.colors', {}),
+        visData.rows,
+        props.palettesRegistry,
+        services.fieldFormats,
+        syncColors
+      ),
+    [
+      bucketColumns,
+      visParams,
+      props.uiState,
+      props.palettesRegistry,
+      visData.rows,
+      services.fieldFormats,
+      syncColors,
+    ]
+  );
+  const config = useMemo(() => getConfig(visParams, chartTheme), [chartTheme, visParams]);
+  const tooltip: TooltipProps = {
+    type: visParams.addTooltip ? TooltipType.Follow : TooltipType.None,
+  };
+  const legendPosition = useMemo(() => visParams.legendPosition ?? Position.Right, [
+    visParams.legendPosition,
+  ]);
+
+  const legendColorPicker = useMemo(
+    () =>
+      getColorPicker(
+        legendPosition,
+        setColor,
+        bucketColumns,
+        visParams.palette.name,
+        visData.rows,
+        props.uiState,
+        visParams.distinctColors
+      ),
+    [
+      legendPosition,
+      setColor,
+      bucketColumns,
+      visParams.palette.name,
+      visParams.distinctColors,
+      visData.rows,
+      props.uiState,
+    ]
+  );
+
+  const splitChartColumnAccessor = visParams.dimensions.splitColumn
+    ? getSplitDimensionAccessor(
+        services.fieldFormats,
+        visData.columns
+      )(visParams.dimensions.splitColumn[0])
+    : undefined;
+  const splitChartRowAccessor = visParams.dimensions.splitRow
+    ? getSplitDimensionAccessor(
+        services.fieldFormats,
+        visData.columns
+      )(visParams.dimensions.splitRow[0])
+    : undefined;
+
+  const splitChartDimension = visParams.dimensions.splitColumn
+    ? visData.columns[visParams.dimensions.splitColumn[0].accessor]
+    : visParams.dimensions.splitRow
+    ? visData.columns[visParams.dimensions.splitRow[0].accessor]
+    : undefined;
+
+  return (
+    <div className="pieChart__container" data-test-subj="visTypePieChart">
+      <div className="pieChart__wrapper">
+        <LegendToggle
+          onClick={toggleLegend}
+          showLegend={showLegend}
+          legendPosition={legendPosition}
+        />
+        <Chart size="100%">
```

Comment:
I found that we should reduce and limit the size of the rendered pie in two places:
- when rendering the pie in the dashboard, the pie doesn't have margins (Lens has a bit of margin, at least on the bottom). The margin will help avoid external labels conflicts with the Hide/Show legend button
See below Lens on the Right, this implementation on the Left 
<img width="1268" alt="Screenshot 2021-04-29 at 14 29 04" src="https://user-images.githubusercontent.com/1421091/116559231-7ef4d200-a900-11eb-8399-129c6cf362de.png">
This was the previous vislib implementation compared with Lens
<img width="1266" alt="Screenshot 2021-04-29 at 14 30 19" src="https://user-images.githubusercontent.com/1421091/116560987-2e7e7400-a902-11eb-9709-37cd1e6aedd1.png">

- Within the editor, the pie is too big, we should limit its parent size to a maximum, to avoid having this big donut on the screen that doesn't provide much more information then the normal one. This also helps a bit avoiding the label truncation when rendered outside
<img width="1908" alt="Screenshot 2021-04-29 at 14 40 12" src="https://user-images.githubusercontent.com/1421091/116559288-93d16580-a900-11eb-811a-72a0d257a50a.png">

Reply:
I agree with @markov00 that the pie looks way too big in the editor. Not because of no room for linked labels; simply, there's a healthy maximal size of the thing. It'd be great to edit it in approximately the size it'll be on the dashboard. That info may not be available, so in its absence, a reasonable maximum would be something like a bounding box that's a third or maybe a half of each of the dimensions (ie. a ninth or a quarter of the screen area). It's unlikely that pies will be bigger than that on the dashboard, and it's best to see it close to the final state, due to text layouting.

However I think that generally, linked labels should not be the default choice, and in-slice labels should be preferred. At least, the user should have clarity if they want to make something like the small, maybe thin donut ring example with exclusively linked labels (bottom left of Marco's example) or an area covering one with in-sector fill labels, with zero linked labels as on the top and bottom right.

So, in summary, I think
- there should be some maximum size to the pie, even when editing, to let the user not be misguided about how it'll look on the dashboard, and huge pies don't make much sense anyway (unlike eg. huge treemaps)
- default: all fill labels, with zero linked labels
- optional, if the user chooses so: much smaller disk, and linked labels exclusively (even then, strongly limit max. label count)
- if the user really really wants, mixed fill + link label can be enabled for the PIE only
- for DONUT, or SUNBURST, I propose, never enable mixed fill + link label - it generally looks awkward and poorly designed
- note: linked labels are especially problematic with a horizontal arrangement of small multiples

@markov00 what do you think?

Reply:
@markov00 I have added a padding on the container and now in a dashboard Lens and new pie are rendered exactly the same
![image](https://user-images.githubusercontent.com/17003240/117108533-01134980-ad8c-11eb-8278-face759bf62c.png)

I have also given the container a max-width of 1024px. This size is under discussion of course
![image](https://user-images.githubusercontent.com/17003240/117109831-ecd04c00-ad8d-11eb-89d4-bb434d7d249e.png)

@ryankeairns wdyt? https://github.com/elastic/kibana/pull/83929/commits/a0087eca7dadb08ec3361a7d69c9d1a2e12238a6

@monfera about your comments I will wait for Marco's answer first

Reply:
Update: When I added a max-width to the parent container the legend position is not where we want
![image](https://user-images.githubusercontent.com/17003240/118448637-ebdadb00-b6fa-11eb-968d-7c25024bfa3a.png)

So I am going to revert it. Do you have any idea how we could change the max-width of the chart without affecting the legend position?